server-mirror-git/src/main/java/com/linecorp/centraldogma/server/internal/mirror/SshGitMirror.java

```diff
@@ -87,52 +88,42 @@ final class SshGitMirror extends AbstractGitMirror {
 
     @Override
     protected void mirrorLocalToRemote(File workDir, int maxNumFiles, long maxNumBytes) throws Exception {
-        SshClient sshClient = null;
-        ClientSession session = null;
-        try (GitWithAuth git = openGit(workDir)) {
-            sshClient = createSshClient();
-            session = createSession(sshClient, git);
-            final DefaultGitSshdSessionFactory sessionFactory =
-                    new DefaultGitSshdSessionFactory(sshClient, session);
-            mirrorLocalToRemote(git, maxNumFiles, maxNumBytes, sessionFactory::configureCommand);
+        final SshClient sshClient = createSshClient();
+        final URIish remoteUri = remoteUri();
+        final ClientSession session = createSession(sshClient, remoteUri);
+        final DefaultGitSshdSessionFactory sessionFactory =
+                new DefaultGitSshdSessionFactory(sshClient, session);
+        try (GitWithAuth git = openGit(workDir, remoteUri, sessionFactory::configureCommand)) {
+            mirrorLocalToRemote(git, maxNumFiles, maxNumBytes);
         } finally {
             try {
-                if (session != null) {
-                    session.close(true);
-                }
+                session.close(true);
             } finally {
-                if (sshClient != null) {
-                    sshClient.stop();
-                }
+                sshClient.stop();
             }
         }
     }
 
     @Override
     protected void mirrorRemoteToLocal(File workDir, CommandExecutor executor,
                                        int maxNumFiles, long maxNumBytes) throws Exception {
-        SshClient sshClient = null;
-        ClientSession session = null;
-        try (GitWithAuth git = openGit(workDir)) {
-            sshClient = createSshClient();
-            session = createSession(sshClient, git);
-            final DefaultGitSshdSessionFactory sessionFactory =
-                    new DefaultGitSshdSessionFactory(sshClient, session);
-            mirrorRemoteToLocal(git, executor, maxNumFiles, maxNumBytes, sessionFactory::configureCommand);
+        final SshClient sshClient = createSshClient();
+        final URIish remoteUri = remoteUri();
+        final ClientSession session = createSession(sshClient, remoteUri);
```

Comment:
We need to close the `sshClient` even if an exception is raised while creating `ClientSession`.